doc/source/tune/user-guide.rst

```diff
@@ -295,49 +295,146 @@ When running a hyperparameter search, Tune can automatically and periodically sa
+        checkpoint_score_attr="max-auc",
+        keep_checkpoints_num=5,
+
+        # a very useful trick! this will resume from the last run specified by
+        # sync_config (if one exists), otherwise it will start a new tuning run
+        resume="AUTO",
+    )
 
-    tune.run(train_func)
+In this example, checkpoints will be saved:
 
-In this example, checkpoints will be saved by training iteration to ``local_dir/exp_name/trial_name/checkpoint_<step>``.
+* **Locally**: not saved! Nothing will be sync'd to the driver (your laptop) automatically
+* **S3**: ``s3://my-checkpoints-bucket/path/my-tune-exp/<trial_name>/checkpoint_<step>``
+* **On head node**: ``~/ray-results/my-tune-exp/<trial_name>/checkpoint_<step>`` (but only for trials done on that node, due to ``sync_to_driver=False``)
+* **On workers nodes**: ``~/ray-results/my-tune-exp/<trial_name>/checkpoint_<step>`` (but only for trials done on that node)
 
-You can restore a single trial checkpoint by using ``tune.run(restore=<checkpoint_dir>)`` By doing this, you can change whatever experiments' configuration such as the experiment's name:
+If your run stopped for any reason (finished, errored, user CTRL+C), you can restart it any time by running the script above again -- note with ``resume="AUTO"``, it will detect the previous run so long as the ``sync_config`` points to the same location.
+
+If, however, you prefer not to use ``resume="AUTO"`` (or are on an older version of Ray) you can resume manaully:
 
 .. code-block:: python
 
     # Restored previous trial from the given checkpoint
     tune.run(
-        "PG",
-        name="RestoredExp", # The name can be different.
-        stop={"training_iteration": 10}, # train 5 more iterations than previous
-        restore="~/ray_results/Original/PG_<xxx>/checkpoint_5/checkpoint-5",
-        config={"env": "CartPole-v0"},
+        # our same trainable as before
+        durable_trainer_with_params,
+
+        # The name can be different from your original name
+        name="my-tune-exp-restart",
+
+        # our same config as above!
+        restore=sync_config,
     )
 
+A simple local/rsync checkpointing example
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Local or rsync checkpointing can be a good option if:
+
+1. You want to tune on a single laptop Ray cluster
+2. You aren't using Ray on Kubernetes (rsync doesn't work with Ray on Kubernetes)
+3. You don't want to use S3
+
+Let's take a look at an example:
+
+.. code-block:: python
+
+    import ray
+    from ray import tune
+    from your_module import my_trainable
+
+    ray.init(address="<cluster-IP>:<port>")  # set `address=None` to train on laptop
+
+    # configure how checkpoints are sync'd to the scheduler/sampler 
+    sync_config = tune.syncConfig(sync_to_driver=True)
+
+    # with_parameters() ensures when we checkpoint, weights are saved (not just hyperparameters)
+    durable_trainer_with_params = tune.with_parameters(my_trainable)
```

Comment:
Ah this is not correct - let's not use `with_parameters` here at all

Generally, `with_parameters` is used to e.g. re-use data across trainables. It saves parameters in the object store to be re-used, similarly to `functools.partial` - we don't use this ere, so let's just directly pass the trainable below